### PR TITLE
[TwigBundle] register template paths for overridden bundles

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | FriendsOfSymfony/FOSUserBundle#2376, FriendsOfSymfony/FOSUserBundle#2378
| License       | MIT
| Doc PR        | 

Previously, paths for bundles overriding a parent bundle wouldn't have been registered with the namespace of the parent bundle. I am not completely sure if this is the best solution, so I didn't invest time to write tests confirming the fix. Will do that if we agree on this approach.

### CODE REVIEW

**Comment by ro0NL** [comment]
Duplicates #19586 i believe, ref #18563

The magic bundle initialization approach is under discussion i guess :) Still think #19646 was a good approach.

File: src/Symfony/Bundle/TwigBundle/DependencyInjection/TwigExtension.php [hunk 1]
```diff
@@ -89,14 +89,39 @@ public function load(array $configs, ContainerBuilder $container)
         }
 
         // register bundles as Twig namespaces
+        $pathsByBundle = array();
+        $overriddenPathsByBundle = array();
         foreach ($container->getParameter('kernel.bundles') as $bundle => $class) {
-            if (is_dir($dir = $container->getParameter('kernel.root_dir').'/Resources/'.$bundle.'/views')) {
-                $this->addTwigPath($twigFilesystemLoaderDefinition, $dir, $bundle);
+            $reflection = new \ReflectionClass($class);
+            $bundleInstance = $reflection->newInstanceWithoutConstructor();
```
Comment (ro0NL) [inline src/Symfony/Bundle/TwigBundle/DependencyInjection/TwigExtension.php]: It wont help virtual bundles really, imo. we need to be rock solid on this.

Comment (xabbuh) [inline src/Symfony/Bundle/TwigBundle/DependencyInjection/TwigExtension.php]: What do you mean with a virtual bundle?

Comment (ro0NL) [inline src/Symfony/Bundle/TwigBundle/DependencyInjection/TwigExtension.php]: `new VirtualBundle($name, $path, BundleInterface $parent)`

**Comment by xabbuh** [comment]
Thanks for pointing me to #19586. I must have completely missed that PR. So I am closing here in favour of #19586.

**Comment by Jurigag** [comment]
Works for me. When this will be merged to 2.8 ?

**Comment by stof** [comment]
@Jurigag this PR has not been merged at all, but closed in favour of another one

**Comment by Jurigag** [comment]
@stof oh, right i missed @xabbuh comment

**Pull request closed without merging**